server-side/webSocket.js

```diff
@@ -0,0 +1,40 @@
+import { WebSocketServer } from 'ws';
+import mongoose from 'mongoose';
+import { connectMongo } from './config/db.js'; // Assuming this connects to MongoDB
+import { EventEmitter } from 'events';
+import { getCountUnreadMessages } from './controllers/message.controller.js';
+import dotenv from 'dotenv';
+dotenv.config();
+
+const wss = new WebSocketServer({ port: 8080 });
+
+connectMongo();
+
+const eventEmitter = new EventEmitter(); 
+
+wss.on('connection', (ws) => {
+  console.log('Client connected');
+  ws.on('message', async (message) => {
+    ws.userId = message.toString();
```

Comment:
you should add switch-case to message.type,  ws can treat multiple types of messages